benchmarking/src/lib.rs

```diff
@@ -1358,3 +1372,53 @@ macro_rules! list_benchmark {
 		$list.push(pallet_benchmarks)
 	)
 }
+
+/// Callback for `define_benchmarks` to call `list_benchmark`.
+#[macro_export]
+macro_rules! cb_list_benchmarks {
+    // anchor
+	( $list:ident, $extra:ident, [ $name:path, $( $location:tt )* ] ) => {
+        list_benchmark!( $list, $extra, $name, $( $location )* );
+	};
+    // recursion tail
+	( $list:ident, $extra:ident, [ $name:path, $( $location:tt )* ] $([ $names:path, $( $locations:tt )* ])+ ) => {
+        cb_list_benchmarks!( $list, $extra, [ $name, $( $location )* ] );
+        cb_list_benchmarks!( $list, $extra, $([ $names, $( $locations )* ])+ );
+    }
+}
+
+/// Defines pallet configs that `add_benchmarks` and `list_benchmarks` use.
+/// Should be preferred instead of having a repetitive list of configs
+/// in `add_benchmark` and `list_benchmark`.
+#[macro_export]
+macro_rules! define_benchmarks {
+    ( $([ $names:path, $( $locations:tt )* ])* ) => {
+		/// Calls `list_benchmark` with all configs from `define_benchmarks`
+		/// and passes the first two parameters on.
+		///
+		/// Use as:
+		/// ```ignore
+		/// list_benchmarks!(list, extra);
+		/// ```
+		#[macro_export]
+        macro_rules! list_benchmarks {
```

Comment:
indent level seems wrong

Reply:
hmmm, looks right again in PR conversation page, probably a github front end bug

Reply:
it is tab vs space thing